test.js

```diff
@@ -0,0 +1,124 @@
+const app = require('express')()
+const bodyParser = require('body-parser')
+const multer = require('multer') // v1.0.5
+const upload = multer() // for parsing multipart/form-data
+
+app.use(bodyParser.json()) // for parsing application/json
+app.use(bodyParser.urlencoded({ extended: true })) // for parsing application/x-www-form-urlencoded
+
+app.post('/profile', upload.array(), function (req, res, next) {
+  console.log(req.body)
+  res.json(req.body)
+})
+
+const port = 3001;
+
+app.get('/', (req, res) => res.send('Hello World!'));
+
+
+class Treenode {
+
+  constructor(label) {
+    this.id = (++Treenode.id_gen).toString()
+    // Treenode.lineage[this.id]=[label]
+    this.label =  label
+    this.parent_id = null
+    // this.parent_label = null
+    this.children = []
+    // Treenode.label_id[this.id] = label
+  }
+
+  add_child(child){
+    this.children.push(child)
+    child.parent_id = this.id
+  }
+
+}
+
+function find_id(iter, parent_id) {
+  if (iter.id === parent_id) {
+    return iter
+  }
+  else {
+    if (iter.children.length > 0) {
+      for (var i = 0; i < iter.children.length; i++){
+        retVal = find_id(iter.children[i], parent_id)
+        if (retVal !== null) {
+          return retVal
+        }
+      }
+    }
+  }
+  return null
+}
+
+Treenode.id_gen = 0;
+Treenode.label_id = {};
+
+root = new Treenode('root')
+ant = new Treenode('ant')
+bear = new Treenode('bear')
+cat = new Treenode('cat')
+dog = new Treenode('dog')
+elephant = new Treenode('elephant')
+frog = new Treenode('frog')
+
+root.add_child(ant)
+root.add_child(bear)
+bear.add_child(cat)
+bear.add_child(dog)
+dog.add_child(elephant)
+root.add_child(frog)
+
+
+var str = JSON.stringify(root, null, 5);
+
+app.get('/api/tree', function (req, res) {
+  res.send(root);
+});
+
+app.post('/api/tree', function (req, res) {
```

Comment:
I have no idea what I need to pass in to use this api / what your request.body is supposed to look like. You can make this better by documenting in a doc comment, or even better by defining another class that defines the structure of the object. We're able to resolve this in typescript by giving the request.body an explicit type so it lets developers know what fields are expected

Reply:
oh, you do have a comment, its not apparent,
see some examples here
https://jsdoc.app/about-getting-started.html